devito/function.py

```diff
@@ -406,12 +412,20 @@ def _indices(cls, **kwargs):
         """
         save = kwargs.get('save', None)
         grid = kwargs.get('grid', None)
+        time_dim = kwargs.get('time_dim', None)
+
+        if time_dim is not None:
+            assert(isinstance(time_dim, TimeDimension))
```

Comment:
Nitpicking: Could combine with below conditional to save lines and make things easier accessible.

Reply:
Isn't this already caught by the constructor?

Reply:
> Isn't this already caught by the constructor?

indeed

Reply:
Actually, the first line of the constructor calls the `super` constructor which [calls](https://github.com/opesci/devito/blob/5fd034fa2c4a7254ff058b6ff634b1e9f64dfdcc/devito/function.py#L141) this so `_indices()` is called before our own constructor and this was thus required. The check in the constructor is actually a second-level check, just in case.

Reply:
> Could combine with below conditional to save lines

Fixed